xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4542,6 +4542,47 @@ int CVideoPlayer::AddSubtitleFile(const std::string& filename, const std::string
     if (!vobsubidx.empty())
       return AddSubtitleFile(vobsubidx, filename);
   }
+  if (ext == ".sup")
+  {
```

Comment:
Nothing wrong with the logic I think, pretty much similar to other sub types in the same file. Wondering about code duplication though, maybe this can be refactored to group similar/identical code somewhere else?